Modix.Common/Messaging/MessagingOptions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Modix.Common.Messaging
+{
+    /// <summary>
+    /// Defines configuration options for the behavior of the messaging system.
+    /// </summary>
+    public class MessagingOptions
+    {
+        /// <summary>
+        /// The amount of time to wait for a <see cref="IMessageDispatcher.Dispatch{TNotification}(TNotification, TimeSpan?)"/> operation to complete,
+        /// when no explicit timeout is specified.
+        /// </summary>
+        public TimeSpan DispatchTimeout { get; set; }
```

Comment:
Probably a bit of an edge case, but would it be worthwhile to add a default value of a minute or something here? Just to be extra sure the bot doesn't die on startup if the appsettings.json isn't there for some reason.

Reply:
It should actually be either nullable, or `Timeout.InfiniteTimeSpan` I suppose. Otherwise it'll be a "soft" failure in that all operations just immediately timeout, cause it defaults to `Timeout.Zero`.